exist-core/src/main/java/org/exist/xquery/functions/array/ArrayFunction.java

```diff
@@ -342,11 +339,29 @@ public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathExce
                         }
                         return array.subarray(start - 1, end);
                     case REMOVE:
-                        final int rpos = ((IntegerValue) args[1].itemAt(0)).getInt();
-                        if (rpos < 1 || rpos > array.getSize()) {
-                            throw new XPathException(this, ErrorCodes.FOAY0001, "Index of item to remove (" + rpos + ") is out of bounds");
+                        // Handle empty sequence
+                        if (args[1].isEmpty()) {
+                            return array;
                         }
-                        return array.remove(rpos - 1);
+
+                        // Get and reverse sort parameters
+                        List<Integer> positions = new ArrayList<>();
```

Comment:
As the size of the positions is known in advance, it would be more performant to use a fixed size array here and then call Arrays.sort.